nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -943,49 +952,122 @@ public List<ExpressionTree> getReturnExpressions() {
-    }
-    Symbol argumentSymbol = ASTHelpers.getSymbol(argument);
-    if (argumentSymbol == null) {
-      return argumentType;
+      Type exprType, ExpressionTree expr, VisitorState state, @Nullable TreePath path) {
+    if (!shouldRunDataflowForExpression(exprType, expr)) {
+      return exprType;
     }
     TreePath currentPath = path != null ? path : state.getPath();
-    // We need a TreePath whose leaf is the argument expression, as the calls to `getNullness` /
+    // We need a TreePath whose leaf is the expression, as the calls to `getNullness` /
     // `getNullnessFromRunning` below return the nullness of the leaf of the path.
-    // Just appending argument to currentPath is a bit sketchy, as it may not actually be the valid
-    // tree path to the argument.  However, all we need the path for (beyond the leaf) is to
+    // Just appending expr to currentPath is a bit sketchy, as it may not actually be the valid
+    // tree path to the expression.  However, all we need the path for (beyond the leaf) is to
     // discover the enclosing method/lambda/initializer, and for that purpose this should be
     // sufficient.
-    TreePath argumentPath = new TreePath(currentPath, argument);
-    TreePath enclosingPath = NullabilityUtil.findEnclosingMethodOrLambdaOrInitializer(argumentPath);
+    TreePath exprPath = new TreePath(currentPath, expr);
+    TreePath enclosingPath = NullabilityUtil.findEnclosingMethodOrLambdaOrInitializer(exprPath);
     if (enclosingPath == null) {
-      return argumentType;
+      return exprType;
+    }
+    // If the expression is inside a lambda, we need to ensure that the environment mapping for
+    // the lambda is set up so that dataflow analysis can be run within the lambda body
+    boolean enclosingIsLambda = enclosingPath.getLeaf() instanceof LambdaExpressionTree;
+    if (enclosingIsLambda) {
+      TreePath methodEnclosingLambda =
+          NullabilityUtil.findEnclosingMethodOrLambdaOrInitializer(enclosingPath);
+      if (methodEnclosingLambda == null) {
+        return exprType;
+      } else {
+        updateEnvironmentMappingForLambda(state, methodEnclosingLambda, enclosingPath);
+      }
     }
     Nullness refinedNullness;
     AccessPathNullnessAnalysis nullnessAnalysis = analysis.getNullnessAnalysis(state);
     if (nullnessAnalysis.isRunning(enclosingPath, state.context)) {
       // dataflow analysis is already running, so just get the current dataflow value for the
       // argument
-      refinedNullness = nullnessAnalysis.getNullnessFromRunning(argumentPath, state.context);
+      refinedNullness = nullnessAnalysis.getNullnessFromRunning(exprPath, state.context);
     } else {
-      refinedNullness = nullnessAnalysis.getNullness(argumentPath, state.context);
+      refinedNullness = nullnessAnalysis.getNullness(exprPath, state.context);
     }
+    // TODO if we did the dataflow analysis for a lambda, always remove the analysis result from the
+    //  cache for safety
```

Comment:
Do we have a test case we can add that shows the checker failing in some way due to the lack of this invalidation? If easy to do, we should add the suppressed test, maybe with the loop mentioned in the description for the PR. Not sure how hard is to trigger the incorrect behavior here, though...

Note that I am not asking for a fix in this PR, just a code-level unit test documentation of what the issue looks like :)